### PR TITLE
fix metro statefuncs in datasources, add metro statefunc to ip resources

### LINKED ISSUE #125: Internal Validation errors after sdk v2 merge
```
        Error: InternalValidate
        
        Internal validation of the provider failed! This is always a bug
        with the provider itself, and not a user issue. Please report
        this bug:
        
        5 errors occurred:
        	* data source metal_facility: metro: StateFunc is extraneous, value should
        just be changed before setting on computed-only field
        	* data source metal_device: metro: StateFunc is extraneous, value should just
        be changed before setting on computed-only field
        	* data source metal_reserved_ip_block: metro: StateFunc is extraneous, value
        should just be changed before setting on computed-only field
        	* data source metal_connection: metro: StateFunc is extraneous, value should
        just be changed before setting on computed-only field
        	* data source metal_spot_market_request: metro: StateFunc is extraneous,
        value should just be changed before setting on computed-only field

```

### PR DESCRIPTION
**Pull request by t0mk**
Fixes internal validations errors caused by metro lowercasing and sdk-v2 upgrade, fixes #125 

Signed-off-by: Tomas Karasek <tom.to.the.k@gmail.com>

### CODE REVIEW

**Comment by t0mk** [comment]
@displague please check this, it would be good to have it in before `metal_port` merge.

File: metal/datasource_metal_connection.go [hunk 1]
```diff
@@ -162,7 +161,7 @@ func dataSourceMetalConnectionRead(d *schema.ResourceData, meta interface{}) err
 		d.Set("facility", conn.Facility.Code)
 	}
 	if conn.Metro != nil {
-		d.Set("metro", conn.Metro.Code)
+		d.Set("metro", toLower(conn.Metro.Code))
```
Comment (displague) [inline metal/datasource_metal_connection.go:164]: perhaps we should just use `strings.ToLower` and avoid the `interface{}` wrapping.

File: metal/datasource_metal_facility.go [hunk 2]
```diff
@@ -59,7 +58,7 @@ func dataSourceMetalFacilityRead(d *schema.ResourceData, meta interface{}) error
 				"features": f.Features,
 				"metro": func(d *schema.ResourceData, k string) error {
 					if f.Metro != nil {
-						return d.Set(k, f.Metro.Code)
+						return d.Set(k, toLower(f.Metro.Code))
```
Comment (displague) [inline metal/datasource_metal_facility.go:61]: same as https://github.com/equinix/terraform-provider-metal/pull/126/files#r650953584

I think `toLower` makes sense for the statefuncs, but it adds indirection in the `d.Set` arguments.

Comment (t0mk) [reply metal/datasource_metal_facility.go]: OK, I changed toLower to strings.ToLower in relevant places in the code

**Pull request merged**